src/Pact/Repl/Lib.hs

```diff
@@ -270,6 +270,10 @@ replDefs = ("Repl",
       [LitExample "(env-dynref fungible-v2 coin)"]
       ("Substitute module IMPL in any dynamic usages of IFACE in typechecking and analysis. " <>
        "With no arguments, remove all substitutions.")
+     ,defZRNative "env-in-repl" envSimulateNotRepl
+       (funType tTyString [("in-repl", tTyBool)])
+       [LitExample "(env-in-repl true)"]
+       "Set the in-repl flag, to potentially observe behaviors outside of the repl"
```

Comment:
Same note here. This needs to be documented much better.

Reply:
This line is the one that generates the other doc btw.

> What does this do, and why would I set something called "in-repl" to true in order to observe behaviors "out of repl" 

I'm not picky. What would you have me name it?

Reply:
Is a description like this accurate?

```
"`in-repl` is true by default in repl sessions and false otherwise. Manually set `in-repl` to false if you need to simulate non-repl behavior while in a repl workflow, for example to disable repl-only builtin functions."
```

I'm tempted to say `in-repl` could be used to turn repl functions off and on again. But since `in-repl` is a repl builtin, won't it be unavailable after setting `(in-repl false)`? So it's kind of a one-way street? If it _is_ a one-way street, then you don't need the boolean parameter and can just call the method `disable-repl`?

Reply:
>  But since in-repl is a repl builtin, won't it be unavailable after setting (in-repl false)? So it's kind of a one-way street? If it is a one-way street, then you don't need the boolean parameter and can just call the method disable-repl?

That would render it useless for testing. `expect` and `expect-failure` would no longer resolve.

Reply:
I like Greg's description text. It's pretty clear about why this flag exists, and what you might need it for. However, it doesn't actually simulate "not being in a REPL": it only does so for evaluation behavior. Maybe `env-eval-within-repl`?

Reply:
I changed it to `env-simulate-onchain`.